File: VB/PersonalOrganizer/Common/DataModel/DesignTimeUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Collections.Generic
Namespace Common.DataModel
    Public Class DesignTimeUnitOfWork
        Implements IUnitOfWork
        Public Shared ReadOnly Instance As IUnitOfWork = New DesignTimeUnitOfWork()
        Private Sub SaveChanges() Implements IUnitOfWork.SaveChanges
        End Sub
        Private Function GetState(ByVal entity As Object) As EntityState Implements IUnitOfWork.GetState
            Return EntityState.Detached
        End Function
        Private Sub Update(ByVal entity As Object) Implements IUnitOfWork.Update
        End Sub
        Private Sub Detach(ByVal entity As Object) Implements IUnitOfWork.Detach
        End Sub
    End Class
End Namespace

File: CS/PersonalOrganizer/Common/DataModel/DesignTimeUnitOfWork.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace PersonalOrganizer.Common.DataModel {
    public class DesignTimeUnitOfWork : IUnitOfWork {
        public static readonly IUnitOfWork Instance = new DesignTimeUnitOfWork();

        void IUnitOfWork.SaveChanges() { }
        EntityState IUnitOfWork.GetState(object entity) {
            return EntityState.Detached;
        }
        void IUnitOfWork.Update(object entity) { }
        void IUnitOfWork.Detach(object entity) { }
    }
}